site/public/ts/shared/utils/ThumbnailGenerator.ts

```diff
@@ -24,21 +25,7 @@ export class ThumbnailGenerator {
     public generate(designer: CircuitDesigner): string {
         const all = (<CullableObject[]>designer.getObjects()).concat(designer.getWires());
 
-        // Define world bounding box for empty circuits so that it shows a little bit of grid
-        const bbox = CircuitBoundingBox(all);
-        const min = (all.length == 0) ? (EMPTY_CIRCUIT_MIN) : (bbox.getMin());
-        const max = (all.length == 0) ? (EMPTY_CIRCUIT_MAX) : (bbox.getMax());
-
-        // Center and zoom the camera so everything fits
-        //  with extra padding on sides
-        const center = min.add(max).scale(0.5);
-        const relativeSize = max.sub(min).scale(1/this.size);
-        const zoom = Math.max(relativeSize.x, relativeSize.y) * THUMBNAIL_ZOOM_PADDING_RATIO;
-
-        // Move camera
-        const camera = this.view.getCamera();
-        camera.setPos(center);
-        camera.setZoom(zoom);
+        FitCamera(this.view.getCamera(), all, THUMBNAIL_ZOOM_PADDING_RATIO);
```

Comment:
For 1) we'd also have to change this to use the action, which could easily be done by just changing this to

`FitCamera(this.view.getCamera(), all, THUMBNAIL_ZOOM_PADDING_RATIO).execute();`